core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -303,38 +323,37 @@ private static DataSource createRealmDataSource(RealmConfiguration realmConfig)
                     }
                 }
             }
-            rs = prepStmt.executeQuery();
-            List<String> lst = new ArrayList<String>();
-            while (rs.next()) {
-                String name = rs.getString(1);
-                String domain = rs.getString(2);
-                if (domain != null) {
-                    name = UserCoreUtil.addDomainToName(name, domain);
+            try (ResultSet rs = prepStmt.executeQuery()) {
+                List<String> lst = new ArrayList<String>();
+                while (rs.next()) {
+                    String name = rs.getString(1);
+                    String domain = rs.getString(2);
+                    if (domain != null) {
+                        name = UserCoreUtil.addDomainToName(name, domain);
+                    }
+                    lst.add(name);
                 }
-                lst.add(name);
-            }
-            if (lst.size() > 0) {
-                values = lst.toArray(new String[lst.size()]);
+                if (lst.size() > 0) {
+                    values = lst.toArray(new String[lst.size()]);
+                }
+                return values;
             }
-            return values;
         } catch (SQLException e) {
             String errorMessage = "Using sql : " + sqlStmt + " " + e.getMessage();
             if (log.isDebugEnabled()) {
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
-            DatabaseUtil.closeAllConnections(null, rs, prepStmt);
+        }finally {
+            close(dbConnection);
```

Comment:
Since the connection getting passed as a parameter to this method, it shouldn't close the connection inside this method.

Need to close the connection in the original place where it originally (create and )close the dbConnection.